operator/src/test/java/org/keycloak/operator/testsuite/integration/PodTemplateTest.java

```diff
@@ -222,4 +223,53 @@ public void testPodTemplateIncorrectImagePullSecretsConfig() {
                 });
     }
 
+
+    @Test
+    public void testPodTemplateIfHasDuplicatedProxyEnvVarSet() {
+
+        var keycloakWithPodTemplate = k8sclient.load(getClass()
+                .getResourceAsStream("/correct-with-proxy-set-podtemplate-keycloak.yml"));
+
+        // Act
+        keycloakWithPodTemplate.createOrReplace();
+
+        // Assert
+        Awaitility
+                .await()
+                .ignoreExceptions()
+                .atMost(3, MINUTES).untilAsserted(() -> {
+
+                    Log.info("Getting the first Pod and its first Container");
+
+                    var firstKCContainer = k8sclient
+                            .pods()
+                            .inNamespace(namespace)
+                            .withLabel("app", "keycloak")
+                            .list()
+                            .getItems()
+                            .get(0)
+                            .getSpec()
+                            .getContainers()
+                            .get(0);
+
+                    Log.info("Verifying the 'KC_PROXY' Env Var set");
+
+                    assertThat(firstKCContainer.getEnv()
+                            .stream()
+                            .filter(oneEnvVar -> oneEnvVar.getName().equalsIgnoreCase("KC_PROXY"))
+                            .count()).isEqualTo(1);
+
+                    assertThat(firstKCContainer.getEnv()
+                            .stream()
+                            .anyMatch(oneEnvVar -> oneEnvVar.getValue() != null && oneEnvVar.getValue().equalsIgnoreCase(Constants.DEFAULT_DIST_CONFIG.get("proxy")))
+                    ).isFalse();
+
+                    assertThat(firstKCContainer.getEnv()
+                            .stream()
+                            .anyMatch(oneEnvVar -> oneEnvVar.getValue().equalsIgnoreCase("reencrypt"))
+                    ).isTrue();
```

Comment:
We can simplify this as mentioned above.

Reply:
@andre-nascimento-rh please review @vmuzikar comments

Reply:
Suggested change applied.

Thanks.

Reply:
I believe this is not resolved. Please update the asserts here the same way you updated them in `testConfigInCRTakesPrecedence`.

Reply:
Due to the fact that the Operator does not support overriding or removing `Env Vars` throughout `PodTemplate`, makes no sense to have such test method. Therefore, I'm removing all this new method from `PodTemplateTest` class.

Thanks.